src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -400,20 +422,20 @@ final private SecretKey generateSecretKey() throws NoSuchAlgorithmException {
      * @throws GeneralSecurityException
      */
     @TargetApi(18)
-    final synchronized private SecretKey getSecretKeyFromAndroidKeyStore() throws IOException,
+    final synchronized private SecretKey getSecretKeyFromAndroidKeyStore(Context context) throws IOException,
             GeneralSecurityException {
 
         // Loading file and unwrapping this key is causing performance issue.
-        if (sSecretKeyFromAndroidKeyStore != null) {
-            return sSecretKeyFromAndroidKeyStore;
+        if (mSecretKeyFromAndroidKeyStore != null) {
+            return mSecretKeyFromAndroidKeyStore;
         }
 
         // Store secret key in a file after wrapping
-        File keyFile = new File(mContext.getDir(mContext.getPackageName(), Context.MODE_PRIVATE),
+        File keyFile = new File(context.getDir(context.getPackageName(), Context.MODE_PRIVATE),
```

Comment:
It would be great to have a function - "getKeyFile" and do not copy-paste same logic to format file name several times...